GVFS/GVFS.UnitTests/Service/MacServiceTests.cs

```diff
@@ -0,0 +1,141 @@
+
+            repoRegistry.Verify(
+                r => r.TraceStatus(),
+                Times.Once,
+                $"{nameof(repoRegistry.Object.TraceStatus)} was not called during Service startup");
+        }
+
+        [TestCase]
+        public void RepoRegistryMountsOnlyRegisteredRepos()
+        {
+            Mock<GVFSMountProcess> mountProcessMock = new Mock<GVFSMountProcess>();
+            mountProcessMock.Setup(mp => mp.MountRepository(It.IsAny<string>(), It.IsAny<int>(), It.IsAny<ITracer>()));
+
+            this.CreateTestRepos(this.fileSystem, ServiceDataLocation);
+
+            MockPlatform gvfsPlatform = new MockPlatform();
+            RepoRegistry repoRegistry = new RepoRegistry(
+                this.tracer,
+                this.fileSystem,
+                ServiceDataLocation,
+                gvfsPlatform,
+                mountProcessMock.Object);
+
+            repoRegistry.AutoMountRepos(int.Parse(ActiveUserId));
+
+            mountProcessMock.Verify(
+                mp => mp.MountRepository(
+                    It.Is<string>(repo => repo.Equals(RegisteredRepo)),
+                    It.Is<int>(id => id == int.Parse(ActiveUserId)),
+                    It.IsAny<ITracer>()),
+                Times.Once,
+                $"{nameof(mountProcessMock.Object.MountRepository)} was not called for registered repository");
+        }
+
+        [TestCase]
+        public void MountProcessLaunchedUsingCorrectArgs()
+        {
+            string executable = @"/bin/launchctl";
+            string expectedArgs = $"asuser {int.Parse(ActiveUserId)} /usr/local/vfsforgit/gvfs mount {RegisteredRepo}";
+
+            Mock<GVFSMountProcess.MountLauncher> mountLauncherMock = new Mock<GVFSMountProcess.MountLauncher>();
+            mountLauncherMock.Setup(mp => mp.LaunchProcess(
+                It.IsAny<string>(),
+                It.IsAny<string>(),
+                It.IsAny<string>(),
+                It.IsAny<ITracer>()))
+                .Returns(true);
+
+            GVFSMountProcess mountProcess = new GVFSMountProcess(mountLauncherMock.Object, waitTillMounted: false);
+            mountProcess.MountRepository(RegisteredRepo, int.Parse(ActiveUserId), this.tracer);
+
+            mountLauncherMock.Verify(
+                mp => mp.LaunchProcess(
+                    It.Is<string>(exe => exe.Equals(executable)),
+                    It.Is<string>(args => args.Equals(expectedArgs)),
+                    It.Is<string>(workingDirectory => workingDirectory.Equals(RegisteredRepo)),
+                    It.IsAny<ITracer>()),
+                Times.Once,
+                $"{nameof(mountLauncherMock.Object.LaunchProcess)} was not called for registered repository");
+        }
+
+        private void CreateTestRepos(PhysicalFileSystem fileSystem, string dataLocation)
+        {
+            string repo1 = Path.Combine("mock:", "code", "repo1");
+            string repo2 = Path.Combine("mock:", "code", "repo2");

```

Comment:
Oh, I see it now. What about replacing this with 
```suggestion
            string repo2 = RegisteredRepo;
```
to help clarify the intent. Also maybe rename `RegisteredRepo` to `ExpectedActiveRepoPath` or some such?

Reply:
Replaced and renamed.